File: tournaments/arrayCenter/arrayCenter.js
function arrayCenter(a) {
    const result = [];
    const n = a.length;
    let sum = 0;
    let min = Number.MAX_SAFE_INTEGER;

    for (let i = 0; n > i; ++i) {
        sum += a[i];
        min = Math.min(min, a[i]);
    }

    for (let i = 0; n > i; ++i)
        if (Math.abs(n * a[i] - sum) < n * min)
            result.push(a[i]);

    return result;
}

File: tournaments/arrayCenter/arrayCenter.cpp
std::vector<int> arrayCenter(std::vector<int> a) {

  int n = a.size();
  int suma = a[0],
      mina = a[0];
  for (int i = 1; i < n; i++) {
    suma += a[i];
    mina = std::min(mina, a[i]);
  }

  std::vector<int> b;
  for (int i = 0; i < n; i++) {
    if (std::abs(n * a[i] - suma) < n * mina) {
      b.push_back(a[i]);
    }
  }

  return b;
}
